client/src/App.js

```diff
@@ -114,8 +114,36 @@ export default class NormalDisplay extends React.Component {
         }
     }
 
+    async getJournalUrl() {
+        if (this.isAborted) return;
+        let resp;
+        try {
+            resp = await fetch("/api/get-journal-url", {
+                method: 'GET',
+                headers: {
+                    Authorization: this.state.devToolToken
+                }
+            });
+            if (!resp.ok) {
+                const errorMessage = await this.formatErrorMessage(resp, 'Error setting up event provider');
+                throw new Error(errorMessage);
+            }
+            resp = await resp.json();
+            console.log(`Asset Compute journal url: ${resp.journalUrl}`);
+            this.setState({
+                journalUrl: resp.journalUrl
+            });
+            document.getElementById('run').disabled = false;
```

Comment:
It's more optional than anything else, because this should not happen, BUT... I would verify the journal url is valid before enabling the `run` button.

Do we have to do the enabling here also? I think it would be better to not have UI logic in here and do the UI enablement in the caller if possible.

Reply:
(Checking the url for validity would make sense in the caller then, since an aborted call to get the url would return undefined).

Reply:
thanks! got rid of getJournalUrl call. Merged it with checkEventJournal. so no more enabling/disabling logic.